packages/amplify-cli/src/plugin-helpers/platform-health-check.ts

```diff
@@ -0,0 +1,190 @@
+    name: 'javascript',
+    type: 'frontend',
+    packageName: 'amplify-frontend-javascript',
+    packageVersion: '',
+  },
+  mock: {
+    name: 'mock',
+    type: 'util',
+    packageName: 'amplify-util-mock',
+    packageVersion: '',
+  },
+};
+
+export async function checkPlatformHealth(pluginPlatform: PluginPlatform): Promise<boolean> {
+  const activePlugins = pluginPlatform.plugins;
+  setOfficialPackageVersions();
+  const missingOfficialPlugins: Array<PluginDescription> = [];
+  const mismatchedOfficialPlugins: Array<PluginDescription> = [];
+
+  Object.keys(officialPlugins).forEach((plugin: string) => {
+    const officialPluginDescription = officialPlugins[plugin];
+    if (activePlugins[officialPluginDescription.name]) {
+      let isPackageMatching = false;
+      activePlugins[officialPluginDescription.name].every((pluginInfo: PluginInfo) => {
+        if (isMatching(officialPluginDescription, pluginInfo)) {
+          isPackageMatching = true;
+          return false;
+        }
+        return true;
+      });
+
+      if (!isPackageMatching) {
+        mismatchedOfficialPlugins.push(officialPluginDescription);
+      }
+    } else {
+      missingOfficialPlugins.push(officialPluginDescription);
+    }
+  });
+
+  if (missingOfficialPlugins.length > 0) {
+    console.log(chalk.yellow('The following official plugins are missing or inactive:'));
+    missingOfficialPlugins.forEach((pluginDescription: PluginDescription) => {
+      console.log('    ' + pluginDescription.name);
+    });
+  }
+
+  if (mismatchedOfficialPlugins.length > 0) {
+    console.log(chalk.yellow('The following official plugins have incorrect versions:'));
+    mismatchedOfficialPlugins.forEach((pluginDescription: PluginDescription) => {
+      console.log('    ' + pluginDescription.name);
+    });
+  }
+
+  return missingOfficialPlugins.length === 0 && mismatchedOfficialPlugins.length === 0;
+}
+
+function isMatching(pluginDescription: PluginDescription, pluginInfo: PluginInfo) {
+  let result = pluginDescription.packageName === pluginInfo.packageName && pluginDescription.type === pluginInfo.manifest.type;
+
+  if (result && pluginDescription.packageVersion) {
+    result = pluginDescription.packageVersion === pluginInfo.packageVersion;
+  }
+
+  return result;
+}
+
+function setOfficialPackageVersions() {
+  const packageJsonFilePath = path.normalize(path.join(__dirname, '../../package.json'));
+  const packageJson = readJsonFileSync(packageJsonFilePath);
+
+  const coreVersion = packageJson.version;
+  officialPlugins.core.packageVersion = coreVersion;
+
+  const dependencies: { [key: string]: string } = packageJson.dependencies;
+
+  Object.keys(officialPlugins).forEach((plugin: string) => {
+    const { packageName } = officialPlugins[plugin];
+    if (dependencies[packageName]) {
+      const version = dependencies[packageName];
+      officialPlugins[plugin].packageVersion = version;
+    } else {
+      delete officialPlugins[plugin].packageVersion;
```

Comment:
How can we have an official plugin which is not a direct dependency in the CLI?

Reply:
The platform allows that, but currently no such plugins.